apps/rule-manager/app/controllers/RulesController.scala

```diff
@@ -121,4 +121,21 @@ class RulesController(
           )
     }
   }
+
+  def archive(id: Int): Action[AnyContent] = ApiAuthAction { implicit request =>
+    hasPermission(request.user, PermissionDefinition("manage_rules", "typerighter")) match {
```

Comment:
Not for this PR, but a reminder to self that we should find an abstraction for this check, as it's very common!